core/consensus/aura/src/lib.rs

```diff
@@ -112,32 +112,61 @@ fn inherent_to_common_error(err: RuntimeString) -> consensus_common::Error {
 }
 
 /// A digest item which is usable with aura consensus.
-pub trait CompatibleDigestItem<T: Pair>: Sized {
+pub trait CompatibleDigestItem<U, V>: Sized {
 	/// Construct a digest item which is a slot number and a signature on the
 	/// hash.
-	fn aura_seal(slot_number: u64, signature: Signature<T>) -> Self;
+	fn aura_seal(slot_number: U, signature: V) -> Self;
 
 	/// If this item is an Aura seal, return the slot number and signature.
-	fn as_aura_seal(&self) -> Option<(u64, Signature<T>)>;
+	fn as_aura_seal(&self) -> Option<(U, V)>;
 }
 
-impl<T: Pair, Hash, AuthorityId> CompatibleDigestItem<T> for generic::DigestItem<Hash, AuthorityId, Signature<T>>
-	where T::Signature: Clone
+#[deprecated]
+pub type DeprecatedDigestItem<T> = CompatibleDigestItem<u64, T>;
+
+pub type ModernDigestItem = CompatibleDigestItem<ConsensusEngineId, Vec<u8>>;
+
+pub trait GoodDigestItem<T: Pair>: CompatibleDigestItem<u64, T::Signature> + CompatibleDigestItem<ConsensusEngineId, Vec<u8>> {}
```

Comment:
Please wrap this line.

(see [style guide: 80chars, _120 if must be_](https://wiki.parity.io/Substrate-Style-Guide))